aip/general/0236.md

```diff
@@ -0,0 +1,315 @@
+stored as a nested collection under that Policy resource.
+
+In this AIP, we will use the example of a hypothetical Policy resource called,
+`Policy`. The following is the pattern for the resource name:
+
+`projects/{project}/locations/{location}/policies/{policy}`
+
+The experimental versions of the resource used for preview mode are stored as a
+nested collection under `Policy`. The resource type will follow the convention
+"<resource_name> + Experiment”. The following is the pattern for the experiment:
+
+`/projects/{project}/locations/{location}/policies/{policy}/experiments/{experiment}`
+
+A proto used to represent an experiment **must** contain the following fields:
+
+  1. The required top-level fields for a resource, like `name` and `etag`.
+  2. The Policy that is being tested itself.
+  3. The `preview_metadata` field for preview.
+
+
+```proto
+message PolicyExperiment {
+
+  // The name of the PolicyExperiment.
+  string name = 1;
+ 
+  // The fields used for log evaluation and eventual promotion.
+  Policy policy = 2;
+
+  // Metadata for preview mode only.
+  PreviewMetadata preview_metadata = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+- A new proto **must** be used for experiments with the Policy proto inlined
+  inside.
+  - The inlined Policy field contains any changes to the live policy that are to
+    be tested using preview mode.
+- The name inside the embedded Policy message **must** be the name of the live
+  poliicy. 
+- When the user is ready to promote an experiment, the Policy message field
+  **must** be copied into the live resource and the experiment is deleted.
+  This can be done manually or via a custom method for commit described below.
+- A product **may** support multiple new configurations concurrently in preview
+  for a single live Policy. This is represented using two or more resources in
+  the nested collection.
+  - The number of configurations in preview for a given live policy **may** be
+    capped at a certain number.
+- Cascading deletes **must** occur: if the top-level resource is deleted, all
+  experiments **must** also be deleted.
+
+
+### Metadata
+A `preview_metadata` field is used to track all variables pertinent to a preview.
+ 
+```proto 
+message PreviewMetadata {
+  // State of the previewed resource. Possible values are ACTIVE and INACTIVE, the 
+  // former which indicates that results are being logged downstream.
+  string state = 1;
+
+  // Ifentifying string common to all preview mode logs for the product. Searching 
+  // logs for this string will isolate all logs pertaining to this preview.
+  string log_prefix = 2;
+
+  // The start time of the current preview.
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of the preview.
+  google.protobuf.Timestamp stop_time = 4;
+}
+```
+- `PreviewMetadata` **must** have the fields defined in the proto above and
+  **must** use the same set of values for state.
```

Comment:
"set of values for state, `ACTIVE` and `INACTIVE`"

or

"The value for `state` **must** be `ACTIVE` or `INACTIVE`"